biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -2128,7 +2128,10 @@ && complainAboutMissingVersionRange(packageRef, ee)) {
 		}
 
 		if (isPedantic() && noimports.size() != 0) {
-			warning("Imports that lack version ranges: %s", noimports);
+			warning(
+				"Imports that lack version ranges due to not being found in any bundle on the -buildpath: %s (These could stem from transitive dependencies of non-OSGi jars in your classpath (e.g. from -includeresource). "
+					+ "Adding this dependency to your -buildpath could help bnd adding version information.)",
```

Comment:
Sorry for being late to the party, but for the sake of completeness: they could also stem from transitive dependencies of OSGi bundles on your classpath, if the import version ranges for those bundles are not specified in the Bundle. Eg, if the bundle imports the dependency via Require-Bundle it won't have the import range, or if it uses Import-Package with no import range. I would put "These could stem from transitive dependencies of the jars on your buildpath. Explicitly adding these dependencies to your -buildpath could help bnd find the correct version information"

Reply:
Thanks a lot. I added your suggestion in https://github.com/bndtools/bnd/pull/6765